### ISSUE TITLE
allow --insecure to skip hostname matching

### ISSUE THREAD

**Issue by maxandersen** [home-assistant/home-assistant-cli#58]
if your server is running with duckdns or similar you will get a certificate error if trying to connect directly via the IP - even if accepting all certificates which is what `--insecure` does.

There might be a way to work around this in ssl/requests layer but requires some explicit install of additional dependencies - see https://github.com/urllib3/urllib3/blob/master/src/urllib3/packages/ssl_match_hostname/_implementation.py for details.

```
 import ssl
   
    ssl.match_hostname = lambda cert, hostname: True
```

should be enough if ipaddress and ssl are installed (which depends on python version)

**Comment by hunter-hunter** [comment]
Is there any progress on implementing this feature?

**Comment by maxandersen** [comment]
I haven't been able to find a way to do it reliably across python versions and would need going on websocket level too now. 

PRs Welcome If you know a way around it. 

Otherwise best way forward is to not have broken certs enabled server side :)

**Comment by hunter-hunter** [comment]
Unfortunately that sort of thing is outside of my wheel house but I will definitely keep an eye out for things that I would be able to help with.

The only reason I ask is that I have SSL certs set up for external access but was hoping to be able to just use this on my local network without making all of the requests go out and then come back in.

From a security perspective, (assuming a secure connection) would accessing locally vs across the internet be comparable as far as hass-cli is concerned?

**Comment by maxandersen** [comment]
Have you tried doing it and you get exception ? 

I'm away from home right now but believe using IP directly worked fine; the problem is if I use something like hassio.local instead. 

Try that. 

With respect to hass-cli using remove Vs local it doesn't care - only diff is speed/routing but for me personally I didn't see much if any diference; but they all depend on your network situation.

**Comment by hunter-hunter** [comment]
Yeah I get a security exception when navigating to the local IP address through a web browser but I expect that based on the setup I have currently.

What I'm going to try next is to set up an NGINX proxy that allows local traffic over http but forces external traffic over https which should solve this issue (for me at least).

**Comment by mk-maddin** [comment]
Any progress on this? I am facing exactly this issue here :/
